### PR TITLE
feat: add JoinChainStrategy

### PR DESCRIPTION
**Pull request by Mrreadiness**
Suggestion to [Issue ](https://github.com/sqlalchemy-filterset/sqlalchemy-filterset/issues/11)

### CODE REVIEW

File: sqlalchemy_filterset/strategies.py [hunk 1]
```diff
@@ -41,6 +43,24 @@ def _build_join(self, query: Select, onclause: ColumnElement[bool]) -> Select:
...
+
+class JoinChainStrategy(BaseStrategy):
+    def __init__(
+        self,
+        *chain: RelationJoinStrategy,
+    ) -> None:
+        self.chain = chain
+
+    def filter(self, query: Select, expression: Any) -> Select:
+        for el in self.chain:
```
Comment (RamiAwar) [inline sqlalchemy_filterset/strategies.py:59]: more efficient to use functools.reduce

File: tests/strategies/test_join_chain.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            ),
+        )
+        self.assert_compile(  # type: ignore[no-untyped-call]
+            strategy.filter(select(Item.id), ItemLink.name == "test"),
+            "SELECT item.id FROM item "
+            "JOIN item_to_item_link ON item.id = item_to_item_link.right_id "
+            "JOIN item_link ON item_link.id = item_to_item_link.left_id "
+            "WHERE item_link.name = 'test'",
+            literal_binds=True,
+        )
```
Comment (ymezencev) [inline tests/strategies/test_join_chain.py:29]: Join на MtoM все таки осуждаю)))))
Фильтр по вложенным Join to Join to Join прям классно

File: sqlalchemy_filterset/strategies.py [hunk 3]
```diff
@@ -52,7 +62,21 @@ def _join_if_necessary(self, query: Select) -> Select:
...
+        self,
+        *chain: RelationJoinStrategy,
+    ) -> None:
+        self.chain = chain
+
+    def filter(self, query: Select, expression: Any) -> Select:
+        query = functools.reduce(
+            lambda query, strategy: strategy.apply_join(query), self.chain, query
+        )
+        return query.where(expression)
```
Comment (ymezencev) [inline sqlalchemy_filterset/strategies.py:80]: Maybe rename 
JoinChainStrategy to MultiJoinStrategy
RelationJoinStrategy to JoinStrategy
RelationSubqueryExistsStrategy to ExistsStrategy

as @RamiAwar suggested in https://github.com/sqlalchemy-filterset/sqlalchemy-filterset/issues/11#issuecomment-2219629117

**Review by ymezencev — CHANGES_REQUESTED** [review]
We need to add this to the docs

**Review by RamiAwar — COMMENTED** [review]
lgtm! nice work.

my only comment: should is_outer / is_full be an argument? or its own join type? since it doesn't already exist, we can introduce it however we like.

I think copying sqlalchemy and SQL closely is better as a user experience. When you look at the multi join strategy code you'll more easily understand it if you had JoinStrategy and OuterJoinStrategy (because it resembles written SQL then).

is_full I think can be an argument of OuterJoin on the other hand, but maybe also nicer to be explicit and create another type here too.

**Comment by Mrreadiness** [comment]
@RamiAwar I thought about different classes but now they look overwhelmed. The current interface is the same as [sqlalchemy](https://docs.sqlalchemy.org/en/14/core/selectable.html#sqlalchemy.sql.expression.Select.join), it's more flexible than classes.

**Review by ymezencev — APPROVED** [review]

**Comment by RamiAwar** [comment]
Fair point if sqlalchemy also supports this.

Alrightt

**Review by RamiAwar — APPROVED** [review]

**Review by ArtemTitorenko — APPROVED** [review]

**Pull request merged**